### ISSUE TITLE
Xaml view cannot be found if namespace doesn't match folder path

### ISSUE THREAD

**Issue by mat1jaczyyy** [AvaloniaUI/Avalonia#2367]
.cs files are unable to find corresponding XAML if the XAML file is placed inside a folder (for organization purposes). This raises an exception when Show is called on an instance of the Window: `System.IO.FileNotFoundException: 'Unable to find view for api.Track'`. (using netcoreapp2.1)

This approach functions properly:
![image](https://user-images.githubusercontent.com/13300194/54158950-0bf01f00-444c-11e9-86ce-397744d4dd8a.png)

The approach I want to achieve throws an error:
![image](https://user-images.githubusercontent.com/13300194/54158986-21654900-444c-11e9-867d-c60c9b605414.png)
![image](https://user-images.githubusercontent.com/13300194/54159081-67221180-444c-11e9-9485-d497cb98d780.png)

This approach also functions properly, but still doesn't achieve nice organization.
![image](https://user-images.githubusercontent.com/13300194/54159013-36da7300-444c-11e9-9ae2-d9096b4017aa.png)

Is there perhaps a way to specify in the .csproj file which .xaml file each .cs should use? I could settle for that for my purposes, but IMO this should work automatically.

**Comment by ahopper** [comment]
your .csproj should have something like
```
<ItemGroup>
    <Compile Update="**\*.xaml.cs">
      <DependentUpon>%(Filename)</DependentUpon>
    </Compile>
    <EmbeddedResource Include="**\*.xaml">
      <SubType>Designer</SubType>
    </EmbeddedResource>
    <EmbeddedResource Include="Assets\*" />
  </ItemGroup>
```
in and it is normal to name the cs files .xaml.cs

**Comment by mat1jaczyyy** [comment]
Hi @ahopper, thanks for your suggestion, but after applying the issue persists. The code is hosted in a repository should anyone want to dive more deeply, https://github.com/mat1jaczyyy/apollo-studio/tree/avalonia-ui

**Comment by ahopper** [comment]
If you use the templates in vs this stuff just works.
this line https://github.com/mat1jaczyyy/apollo-studio/blob/3a25d2b28694721b5344c48eaaa990dab3a036ab/src/api/api.csproj#L49 looks wrong

**Comment by mat1jaczyyy** [comment]
Unfortunate that I had started the backend of the project way before deciding to use Avalonia for UI. If I remove lines 49 to 51, the error still occurs, just the same as if I keep them in.

**Comment by ahopper** [comment]
you need the namespace of the xaml.cs to match its location i.e api.General

**Comment by grokys** [comment]
If you don't want to follow the standard avalonia layout, you can create an instance of `AvaloniaXamlLoader` and call one of the `Load` methods yourself: see https://github.com/AvaloniaUI/Avalonia/blob/master/src/Markup/Avalonia.Markup.Xaml/AvaloniaXamlLoader.cs

**Comment by kekekeks** [comment]
Use `AvaloniaResource` and `x:Class` which are available since 0.8
